tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,381 @@
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_OPEN
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover", SERVICE_CLOSE_COVER, {"entity_id": entity_id}, blocking=True
+    )
+    assert hass.states.get(entity_id).state == STATE_CLOSING
+
+
+def opening_to_stop_feature_mock(feature_mock):
+    """Return an mocked feature which can be updated and stopped."""
+
+    def initial_update():
+        feature_mock.state = 1  # opening
+
+    def stop():
+        feature_mock.state = 2  # manually stopped
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    feature_mock.async_stop = AsyncMock(side_effect=stop)
+
+
+@pytest.mark.parametrize("wrapper", FIXTURES_SUPPORTING_STOP, indirect=["wrapper"])
+async def test_stop(wrapper, hass, config):
+    """Test cover stopping."""
+
+    feature_mock, entity_id = wrapper
+    opening_to_stop_feature_mock(feature_mock)
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_OPENING
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover", SERVICE_STOP_COVER, {"entity_id": entity_id}, blocking=True
+    )
+    assert hass.states.get(entity_id).state == STATE_OPEN
+
+
+@pytest.mark.parametrize("wrapper", ALL_COVER_FIXTURES, indirect=["wrapper"])
+async def test_update(wrapper, hass, config):
+    """Test cover updating."""
+
+    feature_mock, entity_id = wrapper
+
+    def initial_update():
+        feature_mock.current = 29  # inverted
+        feature_mock.state = 2  # manually stopped
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.attributes[ATTR_CURRENT_POSITION] == 71  # 100 - 29
+    assert state.state == STATE_OPEN
+
+
+@pytest.mark.parametrize(
+    "wrapper", ["gatecontroller", "shutterbox"], indirect=["wrapper"]
+)
+async def test_set_position(wrapper, hass, config):
+    """Test cover position setting."""
+
+    feature_mock, entity_id = wrapper
+
+    def initial_update():
+        feature_mock.state = 3  # closed
+
+    def set_position(position):
+        assert position == 99  # inverted
+        feature_mock.state = 1  # opening
+        # feature_mock.current = position
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    feature_mock.async_set_position = AsyncMock(side_effect=set_position)
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_CLOSED
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover",
+        SERVICE_SET_COVER_POSITION,
+        {"entity_id": entity_id, ATTR_POSITION: 1},
+        blocking=True,
+    )  # almost closed
+    assert hass.states.get(entity_id).state == STATE_OPENING
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)